Feature/26/scheduler job

malakai97 opened the pull request:
Resolves #26 

**Abridged** description of commits:

- b929965: Scheduler now expects a time range instead of last_run_time, does not take a logger, and doesn't log.  
- 1f48b74: Removes VolumeActionLogger, now unused.
- ca2359b: Removes the spec/integration values.  If these are needed for integration tests, they should be in a configuration file in the test directory.

aelkiss commented on config/hathitrust_config.rb:
```diff
@@ -77,6 +77,10 @@ def superset_volume_writer
         )
       end
 
+      def profiles
```
this is now duplicated in private and public. It was formerly private, but the datasets_cli needs to know about it unless it's going to special case the subsets vs. superset which I'd rather try to avoid unless necessary

aelkiss commented on lib/datasets/scheduler.rb:
```diff
@@ -9,39 +9,33 @@ class Scheduler
-        SaveJob.new(volume, src_path, volume_writer).enqueue
-        logger.log("save", volume, src_path)
-      end
+        .map {|volume, src_path| SaveJob.new(volume, src_path, volume_writer) }
```
If this is going to work with the expected reporter interface, it needs to end up returning the set of volumes, e.g.

```ruby
      volume_repo.changed_between(time_range.first, time_range.last)
        .select {|volume| filter.matches?(volume) }
        .each { |volume|  SaveJob.new(volume, src_path_resolver.path(volume), volume_writer).enqueue }
```

(could definitely be cleaner, this is just to illustrate). You could also punt and address this separately as part of #56

aelkiss commented on lib/datasets/scheduler.rb:
```diff
@@ -9,39 +9,33 @@ class Scheduler
+        .each {|job| job.enqueue}
     end
 
     def delete
```
I think this previously returned what the reporter expected (a set of volumes), and now it doesn't

aelkiss commented on spec/scheduler_spec.rb:
```diff
@@ -66,13 +62,6 @@ module Datasets
         scheduler.add
       end
 
-      it "logs each volume that satisfies the filter" do
```
If we remove these specs we need to add corresponding specs that it instead returns the volumes that satisfy the filter.

aelkiss commented on config/config.example.yml:
```diff
@@ -3,14 +3,15 @@
+src_parent_dir:  full/src/path
 
 dest_parent_dir:
-  :full: 'spec/integration/datasets/ht_text'
```
please revert to master (see comments on #51)

malakai97 commented:
- Removed duplicate `#profiles` definition.
- Added code+specs to ensure scheduler.add, delete return the enqueued volumes.
- ~Why revert config.example.yml?  Is the integration test consuming it?  The integration tests should read from a config within the spec dir.~ Reverted for now, creating an issue to have tests read config from test dirs.

aelkiss commented:
Why revert config.example.yml: because @grosscol thought it was duplicative with the integration config and wanted a working example config out of the box.
